pandas/tests/indexing/test_timedelta.py

```diff
@@ -40,3 +39,13 @@ def test_list_like_indexing(self, indexer, expected):
                                 dtype="int64")
 
         tm.assert_frame_equal(expected, df)
+
+    def test_string_indexing(self):
+        # GH 16896
+        df = pd.DataFrame({'x': range(3)},
+                          index=pd.to_timedelta(range(3), unit='days'))
+        expected = df.iloc[0]
+        sliced = df.loc['0 days']
+        tm.assert_series_equal(sliced, expected)
+
+        assert df.index.get_loc('0 days') == 0
```

Comment:
There is a check in [test_get_loc](https://github.com/pandas-dev/pandas/blob/25384ba459ba7de9fb9d36821f0a4ae239cc40b2/pandas/tests/indexes/timedeltas/test_timedelta.py#L67)

This line will generate a ValueError in `_get_dtype_type`

'0 days' generates a TypeError instead.